File: backend/API/serializers/tag.py
from rest_framework import serializers
from ..models import Tag
from ..serializers import BaseSerializer


class TagSerializer(BaseSerializer):
    """
    Sérializer du node Tag

    Champs aditionnels :
        - name : nom du tag (obligatoire)
        Relations :
            - interviews : les interviews liées au tag
            - extraits : les extraits liés au tag
    """

    name = serializers.CharField(required=True)

    # Outputs
    interviews = serializers.SerializerMethodField(read_only=True)
    extraits = serializers.SerializerMethodField(read_only=True)

    def __init__(self, *args, **kwargs):
        """
        node : Tag
        """
        super().__init__(Tag, *args, **kwargs)

    def get_interviews(self, tag):
        """
        Renvoie un lien propre vers les interviews :
        """
        return self.get_url("interview-list", kwargs={"tag_uuid": tag.uuid})

    def get_extraits(self, tag):
        """
        Renvoie un lien propre vers les extraits :
        """
        return self.get_url("extrait-list", kwargs={"tag_uuid": tag.uuid})

File: frontend/src/model/tag.js
import Model from "@model/model.js";
import Interview from "@model/interview.js";
import Extrait from "@model/extrait.js";

export default class Tag extends Model {
    #name;
    #interviews;
    #extraits;

    constructor({ uuid, name, interviews, extraits }) {
        super(uuid);
        this.#name = name;
        this.#interviews = interviews;
        this.#extraits = extraits;
    }

    static get endpoint() { return "tags"; }

    get name() { return this.#name; }
    set name(value) { this.#name = this.validateString(value, "name"); }

    async interviews(args) { return this.fetchList(this.#interviews, Interview, args); }

    async extraits(args) { return this.fetchList(this.#extraits, Extrait, args); }

    fromJSON(json) {
        super.fromJSON(json);
        this.#name = json.name;
        this.#interviews = json.interviews;
        this.#extraits = json.extraits;
        return this;
    }

    toJSON(json = {}) {
        json = super.toJSON(json)
        if (this.name) json['name'] = this.name;
        return json;
    }
}
